lib/middleware/website.js

```diff
@@ -72,13 +72,21 @@ exports = module.exports = () =>
       if (err.code !== 'NoSuchKey') throw err;
 
       const key = err.detail.Key;
+      // SPA websites may have url anchors (ie: http://localhost/#my-anchor)
+      const [keyWithoutHash, keyHash] = key.split('#');
```

Comment:
I don't like that this mishandles cases where there are `#` characters in the fragment itself (`/some-key##this-is-weird-but-i-dont-judge`). Assigning a `const fragmentIdx = key.indexOf('#')` and using string slices is probably better.